Cross compiler macOS (macOS AARCH64 -> macOS X86_64) refuses to build X86_64 compiler

I'm running FPCUpDeluxe on macOS 15.4.1 (downgraded XCode and Command Line Tools to 16.2 due to a linker error with 16.3), on a M1 Mac (AARCH64/Apple Silicon), and I'm trying to have it build a FPC compiler for x85_64 (also for macOS).

No matter what I try, it fails.

```
/Applications/Xcode.app/Contents/Developer/usr/bin/make 'FPC=/Users/hans/fpcupdeluxe/fpcsrc/compiler/ppc' OS_TARGET=darwin CPU_TARGET=aarch64 PPC_TARGET=x86_64 EXENAME=ppcrossx64 CROSSBINDIR= BINUTILSPREFIX= CROSSCYCLEBOOTSTRAP=1 CYCLELEVEL=2 cycleclean
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C utils cleanall
/Applications/Xcode.app/Contents/Developer/usr/bin/make 'FPC=/Users/hans/fpcupdeluxe/fpcsrc/compiler/ppc' OS_TARGET=darwin CPU_TARGET=aarch64 PPC_TARGET=x86_64 EXENAME=ppcrossx64 CROSSBINDIR= BINUTILSPREFIX= CROSSCYCLEBOOTSTRAP=1 CYCLELEVEL=2 compiler
/bin/echo "'17801-g2a30ef7aca'" > revision.inc
/Users/hans/fpcupdeluxe/fpcsrc/compiler/ppc -Ur -Xs -O2 -n -Fux86_64 -Fusystems -Fu/Users/hans/fpcupdeluxe/fpcsrc/rtl/units/aarch64-darwin -Fix86_64 -FEx86_64/bin/aarch64-darwin -FUx86_64/units/aarch64-darwin -XR/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -dRELEASE -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap   -DD2025/04/18 -dREVINC -dx86_64 -Fux86 -Fix86 -Sew version.pas
/Users/hans/fpcupdeluxe/fpcsrc/compiler/ppc -Ur -Xs -O2 -n -Fux86_64 -Fusystems -Fu/Users/hans/fpcupdeluxe/fpcsrc/rtl/units/aarch64-darwin -Fix86_64 -FEx86_64/bin/aarch64-darwin -FUx86_64/units/aarch64-darwin -XR/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -dRELEASE -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap   -DD2025/04/18 -dREVINC -dx86_64 -Fux86 -Fix86 -Sew -o/Users/hans/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 pp.pas
-macosx_version_min has been renamed to -macos_version_min
fpcupdeluxe: info: FPC Cross Installer (BuildModuleCustom: FPC): Running make [Compiler Install] (FPC crosscompiler: x86_64-darwin) with CROSSOPT: -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9.
fpcupdeluxe: Executing: /usr/bin/make --directory=/Users/hans/fpcupdeluxe/fpcsrc FPCDIR=/Users/hans/fpcupdeluxe/fpcsrc FPCMAKE=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin/fpcmake PPUMOVE=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin/ppumove PREFIX=/Users/hans/fpcupdeluxe/fpc INSTALL_PREFIX=/Users/hans/fpcupdeluxe/fpc INSTALL_SOURCEDIR=/Users/hans/fpcupdeluxe/fpcsrc INSTALL_BASEDIR=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin INSTALL_UNITDIR=/Users/hans/fpcupdeluxe/fpc/units/x86_64-darwin INSTALL_BINDIR=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin INSTALL_LIBDIR=/Users/hans/fpcupdeluxe/fpc/lib INSTALL_SHAREDDIR=/Users/hans/fpcupdeluxe/fpc/share INSTALL_DATADIR=/Users/hans/fpcupdeluxe/fpc/data INSTALL_DOCDIR=/Users/hans/fpcupdeluxe/fpc/doc INSTALL_EXAMPLEDIR=/Users/hans/fpcupdeluxe/fpc/examples CPU_SOURCE=aarch64 OS_SOURCE=darwin OS_TARGET=darwin CPU_TARGET=x86_64 CROSSINSTALL=1 SYSTEMDIR=/Users/hans/fpcupdeluxe/fpcsrc/rtl/bsd DOSDIR=/Users/hans/fpcupdeluxe/fpcsrc/rtl/unix SYSUTILSDIR=/Users/hans/fpcupdeluxe/fpcsrc/rtl/unix CLASSESDIR=/Users/hans/fpcupdeluxe/fpcsrc/rtl/unix TTHREADINCDIR=/Users/hans/fpcupdeluxe/fpcsrc/rtl/unix CROSSBINDIR=/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin FPCFPMAKE=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin/fpc.sh FPC=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin/fpc.sh compiler_install NOGDBMI=1 REVSTR=17801-g2a30ef7aca REVINC=force OPT=-Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap CROSSOPT=-FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9 (working dir: /Users/hans/fpcupdeluxe/fpcsrc)
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C compiler install
/Applications/Xcode.app/Contents/Developer/usr/bin/make quickinstall auxfilesinstall
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C utils install
/usr/bin/install -m 755 -d /Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin
/usr/bin/install -c -m 755 ppcrossx64 /Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin/ppcrossx64
fpcupdeluxe: info: FPC Cross Installer (BuildModuleCustom: FPC): Running make [Rtl Build] (FPC crosscompiler: x86_64-darwin) with CROSSOPT: -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9.
fpcupdeluxe: Executing: /usr/bin/make --directory=/Users/hans/fpcupdeluxe/fpcsrc FPCDIR=/Users/hans/fpcupdeluxe/fpcsrc FPCMAKE=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin/fpcmake PPUMOVE=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin/ppumove PREFIX=/Users/hans/fpcupdeluxe/fpc INSTALL_PREFIX=/Users/hans/fpcupdeluxe/fpc INSTALL_SOURCEDIR=/Users/hans/fpcupdeluxe/fpcsrc INSTALL_BASEDIR=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin INSTALL_UNITDIR=/Users/hans/fpcupdeluxe/fpc/units/x86_64-darwin INSTALL_BINDIR=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin INSTALL_LIBDIR=/Users/hans/fpcupdeluxe/fpc/lib INSTALL_SHAREDDIR=/Users/hans/fpcupdeluxe/fpc/share INSTALL_DATADIR=/Users/hans/fpcupdeluxe/fpc/data INSTALL_DOCDIR=/Users/hans/fpcupdeluxe/fpc/doc INSTALL_EXAMPLEDIR=/Users/hans/fpcupdeluxe/fpc/examples CPU_SOURCE=aarch64 OS_SOURCE=darwin OS_TARGET=darwin CPU_TARGET=x86_64 CROSSINSTALL=1 SYSTEMDIR=/Users/hans/fpcupdeluxe/fpcsrc/rtl/bsd DOSDIR=/Users/hans/fpcupdeluxe/fpcsrc/rtl/unix SYSUTILSDIR=/Users/hans/fpcupdeluxe/fpcsrc/rtl/unix CLASSESDIR=/Users/hans/fpcupdeluxe/fpcsrc/rtl/unix TTHREADINCDIR=/Users/hans/fpcupdeluxe/fpcsrc/rtl/unix CROSSBINDIR=/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin FPCFPMAKE=/Users/hans/fpcupdeluxe/fpc/bin/aarch64-darwin/fpc.sh FPC=/Users/hans/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 rtl_all NOGDBMI=1 REVSTR=17801-g2a30ef7aca REVINC=force OPT=-Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap CROSSOPT=-FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9 (working dir: /Users/hans/fpcupdeluxe/fpcsrc)
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C rtl all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C darwin all
/Users/hans/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/hans/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap -dx86_64 -dRELEASE -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9  -Us -Sg /Users/hans/fpcupdeluxe/fpcsrc/rtl/bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
genmath.inc(584,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

Notes:
- `/Users/hans/fpcupdeluxe/fpcsrc/compiler/ppcrossx64` exists and runs.
- I tried this with the default setting and with "-WM11.0" - both fail.
- Doing it the other way around (x86_64 -> AARCH64) does work. (using x86_64 FPCUpDeluxe)

Any help or tips would be very much appreciated 😊

FYI, I did try to build FPC 3.3.1 again, and this is the error I got today with "**-WM11.0**":

```
fpcupdeluxe: info: FPC Cross Installer (BuildModuleCustom: FPC): Running make [Rtl Build] (FPC crosscompiler: x86_64-darwin) with CROSSOPT: -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9.
fpcupdeluxe: Executing: /usr/bin/make --directory=/Users/hans/lazarus/fpcupdeluxe/fpcsrc FPCDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc FPCMAKE=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/fpcmake PPUMOVE=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/ppumove PREFIX=/Users/hans/lazarus/fpcupdeluxe/fpc INSTALL_PREFIX=/Users/hans/lazarus/fpcupdeluxe/fpc INSTALL_SOURCEDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc INSTALL_BASEDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin INSTALL_UNITDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/units/x86_64-darwin INSTALL_BINDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin INSTALL_LIBDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/lib INSTALL_SHAREDDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/share INSTALL_DATADIR=/Users/hans/lazarus/fpcupdeluxe/fpc/data INSTALL_DOCDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/doc INSTALL_EXAMPLEDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/examples CPU_SOURCE=aarch64 OS_SOURCE=darwin OS_TARGET=darwin CPU_TARGET=x86_64 CROSSINSTALL=1 SYSTEMDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/bsd DOSDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/unix SYSUTILSDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/unix CLASSESDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/unix TTHREADINCDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/unix CROSSBINDIR=/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin FPCFPMAKE=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/fpc.sh FPC=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 rtl_all NOGDBMI=1 REVSTR=17854-g065a52992a REVINC=force OPT=-Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap CROSSOPT=-FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9 (working dir: /Users/hans/lazarus/fpcupdeluxe/fpcsrc)
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C rtl all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C darwin all
/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap -dx86_64 -dRELEASE -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9  -Us -Sg /Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
genmath.inc(584,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

Same as before, so I'm not sure if posting this is helpful or not. 
Do find it odd that it does work the other way around.



Unfortunately, "**-WM10.15**" does not work for the x86_64 cross compiler either (using **FPCCUpDeluxe AARCH64**) with a different error message related to **genmath.inc**:

```
/usr/bin/install -m 755 -d /Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin
/usr/bin/install -c -m 755 ppcrossx64 /Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/ppcrossx64
fpcupdeluxe: info: FPC Cross Installer (BuildModuleCustom: FPC): Running make [Rtl Build] (FPC crosscompiler: x86_64-darwin) with CROSSOPT: -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9.
fpcupdeluxe: Executing: /usr/bin/make --directory=/Users/hans/lazarus/fpcupdeluxe/fpcsrc FPCDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc FPCMAKE=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/fpcmake PPUMOVE=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/ppumove PREFIX=/Users/hans/lazarus/fpcupdeluxe/fpc INSTALL_PREFIX=/Users/hans/lazarus/fpcupdeluxe/fpc INSTALL_SOURCEDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc INSTALL_BASEDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin INSTALL_UNITDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/units/x86_64-darwin INSTALL_BINDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin INSTALL_LIBDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/lib INSTALL_SHAREDDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/share INSTALL_DATADIR=/Users/hans/lazarus/fpcupdeluxe/fpc/data INSTALL_DOCDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/doc INSTALL_EXAMPLEDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/examples CPU_SOURCE=aarch64 OS_SOURCE=darwin OS_TARGET=darwin CPU_TARGET=x86_64 CROSSINSTALL=1 SYSTEMDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/bsd DOSDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/unix SYSUTILSDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/unix CLASSESDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/unix TTHREADINCDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/unix CROSSBINDIR=/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin FPCFPMAKE=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/fpc.sh FPC=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 rtl_all NOGDBMI=1 REVSTR=17854-g065a52992a REVINC=force OPT=-Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM10.15 -ap CROSSOPT=-FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9 (working dir: /Users/hans/lazarus/fpcupdeluxe/fpcsrc)
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C rtl all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C darwin all
/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM10.15 -ap -dx86_64 -dRELEASE -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.9  -Us -Sg /Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/bsd/system.pp
genmath.inc(584,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
<stdin>:57125:10: error: unknown token in expression
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld23' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld8' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld16' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld59' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld57' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld45' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld7' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld35' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld4' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld54' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld58' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld52' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld53' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld48' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld55' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld49' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Lj9587' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld20' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld29' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld15' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld21' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld41' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld10' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld44' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld32' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld18' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld19' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld11' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld56' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld50' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld12' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld6' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld26' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld14' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld42' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld2' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld30' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld22' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld31' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld40' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld5' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld13' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld36' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld9' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld39' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld33' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld24' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Lj9586' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld28' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld47' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld25' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld34' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld27' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld3' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld38' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld43' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld51' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld17' not defined
        cmpw    $0,
                   ^
<stdin>:57125:10: error: assembler local symbol 'Ld46' not defined
        cmpw    $0,
                   ^
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

Again, odd that FPCUpDeluxe x86_64 has zero issues generating a x86_64 compiler, but FPCUpDeluxe AARCH64 cannot seem to create a x86_64 cross compiler (but that is maybe my lack of knowledge on the topic).

Hope this info helps 😊

Note: 

- Each cross complier build I do by clicking the "install compiler" on the "cross" tab. 
- I've only tested building the FPC compiler in a few scenarios.
- And after each failed build, I always removed ALL files and start again from scratch.
- I did test this with FPC 3.3.1 - as it is stable (I assume).
- I'm using: fpcupdeluxe-**aarch64**-darwin-cocoa 2.4.0gp

Tests:

- Build AARCH64 FPC 3.3.1 (with -WM11.0) - **works**

Next:

- Build x86_64 FPC 3.3.1 (with -WM11.0) Cross Compiler - **fails**

Part of the error output (quite a few "assembler local symbol" errors removed in order to keep things short):

```
genmath.inc(584,5) Fatal: Internal error 2014051001
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
<stdin>:57103:2: error: unknown use of instruction mnemonic without a size suffix
        jmp
        ^
<stdin>:57103:5: error: assembler local symbol 'Ld23' not defined
        jmp
           ^
<stdin>:57103:5: error: assembler local symbol 'Ld8' not defined
```
... etc etc
```
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

Next

- Build AARCH64 compiler FPC 3.3.1, with all defaults setting (so I did not use any "-WM" parameters): **worked**

Next

- Build x86_64 Cross Compiler FPC 3.3.1 with all defaults setting (also without any -WM paramaters): **failed**.

```
/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -ap -dx86_64 -dRELEASE -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.15  -Us -Sg ../bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
genmath.inc(584,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

Next

- Build AARCH64 compiler FPC 3.3.1 (with -WM10.15) **worked**.

Next

- Build x86_64 cross compiler FPC 3.3.1 (with -WM10.15) **failed**.

```
/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM10.15 -ap -dx86_64 -dRELEASE -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.15  -Us -Sg ../bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
genmath.inc(584,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

Next

- Build AARCH64 compiler FPC trunk (3.3.1-17875-g17f7e9f578, with -WM11.0) **worked**.

Next

- Build x86_64 cross compiler FPC 3.3.1 (3.3.1-17875-g17f7e9f578, with -WM11.0) **failed**.

```
/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap -dx86_64 -dRELEASE -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.15  -Us -Sg ../bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
genmath.inc(584,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

Will try fpcupdeluxe-**x86_64**-darwin-cocoa 2.4.0gp next

Will give it a try. 😊

- with your lates fpcupdeluxe-**aarch64**-darwin-cocoa 2.4.0gp
- removed all previous Lazarus/FPC related files

- FPC 3.3.1 AARCH64 with "-WM11.0" - build **works**.
- FPC 3.3.1 x86_64 Cross compiler with "-WM11.0" - build **fails**.

p.s. In both tests did notice in some lines, its still referring to another SDK version (probably not related though). For example:

`Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -**WM10.15**  -Us -Sg ../bsd/system.pp`

```
/usr/bin/install -m 755 -d /Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin
/usr/bin/install -c -m 755 ppcrossx64 /Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/ppcrossx64
fpcupdeluxe: info: FPC Cross Installer (BuildModuleCustom: FPC): Running make [Rtl Build] (FPC crosscompiler: x86_64-darwin) with CROSSOPT: -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.15.
fpcupdeluxe: Executing: /usr/bin/make --directory=/Users/hans/lazarus/fpcupdeluxe/fpcsrc FPCDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc FPCMAKE=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/fpcmake PPUMOVE=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/ppumove PREFIX=/Users/hans/lazarus/fpcupdeluxe/fpc INSTALL_PREFIX=/Users/hans/lazarus/fpcupdeluxe/fpc INSTALL_SOURCEDIR=/Users/hans/lazarus/fpcupdeluxe/fpcsrc INSTALL_BASEDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin INSTALL_UNITDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/units/x86_64-darwin INSTALL_BINDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin INSTALL_LIBDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/lib INSTALL_SHAREDDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/share INSTALL_DATADIR=/Users/hans/lazarus/fpcupdeluxe/fpc/data INSTALL_DOCDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/doc INSTALL_EXAMPLEDIR=/Users/hans/lazarus/fpcupdeluxe/fpc/examples CPU_SOURCE=aarch64 OS_SOURCE=darwin OS_TARGET=darwin CPU_TARGET=x86_64 CROSSINSTALL=1 CROSSBINDIR=/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin FPCFPMAKE=/Users/hans/lazarus/fpcupdeluxe/fpc/bin/aarch64-darwin/fpc.sh FPC=/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 rtl_all NOGDBMI=1 REVSTR=17879-g8e6a4053e3-dirty REVINC=force OPT=-Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap CROSSOPT=-FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.15 (working dir: /Users/hans/lazarus/fpcupdeluxe/fpcsrc)
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C rtl all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C darwin all
/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap -dx86_64 -dRELEASE -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.15  -Us -Sg ../bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
genmath.inc(584,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

- FPC 3.3.1 AARCH64 default settings (without "-WM11.0") - build **works**.
- FPC 3.3.1 x86_64 Cross compiler default settings (without "-WM11.0") - build **fails**.

```
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C rtl all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C darwin all
/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -ap -dx86_64 -dRELEASE -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.15  -Us -Sg ../bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
genmath.inc(584,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

As suspected, based on the previous tests, the "WM" parameter may not be relevant (not knowing the fine details of course)? Even though they come back with different errors.
I'm still confused (again: since I do not know the difference between compiler and cross-compiler) why x86_64 fails as a cross compiler, but just works fine as the regular compiler. Why doesn't the regular x86_64 build fail on "genmath.inc"?

Did the same as before:
- with your lates fpcupdeluxe-aarch64-darwin-cocoa 2.4.0gp
- removed all previous Lazarus/FPC related files

- FPC 3.3.1 AARCH64 with "-WM11.0" - build **works**.
- FPC 3.3.1 x86_64 Cross compiler with "-WM11.0" - build **fails**.

Errors:

```
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C rtl all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C darwin all
/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM11.0 -ap -dx86_64 -dRELEASE -WM11.0 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -dFPC_USE_LIBC  -Us -Sg ../bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
genmath.inc(584,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

- FPC 3.3.1 AARCH64 with "-WM10.15" - build **works**.
- FPC 3.3.1 x86_64 Cross compiler with "-WM10.15" - build **fails**.

```
/Users/hans/lazarus/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/hans/lazarus/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -WM10.15 -ap -dx86_64 -dRELEASE -WM10.15 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -dFPC_USE_LIBC  -Us -Sg ../bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
<stdin>:57111:8: error: assembler local symbol 'Ld23' not defined
.globl  
        ^
<stdin>:57111:8: error: assembler local symbol 'Ld8' not defined
.globl  
        ^
<stdin>:57111:8: error: assembler local symbol 'Ld59' not defined
.globl  
        ^
```
(removed several similar errors)

```
<stdin>:57111:8: error: assembler local symbol 'Ld17' not defined
.globl  
        ^
<stdin>:57111:8: error: assembler local symbol 'Ld46' not defined
.globl  
        ^
genmath.inc(584,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.

ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

Let me know if you want me to test something else 😊 (thank you for your efforts!)

It seems I ran into a similar issue after upgrading from an Intel Mac to Apple Silicon. Installed FPC 3.3.1 and Lazarus 4.4 just fine, but when trying to compile the cross compiler with `-WM11.0` for x86_64 it always fails with:

```
/Users/<xxx>/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -Ur -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/<xxx>/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -ap -dx86_64 -dRELEASE -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.15  -Us -Sg ../bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
genmath.inc(919,7) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

Below the main logs:

```
info: FPC Cross Installer (BuildModuleCustom: FPC): Running make [Rtl Build] (FPC crosscompiler: x86_64-darwin) with CROSSOPT: -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system -WM10.15.
Start of compile error summary.
Compiler error: 2014051001
Start of compile error summary.
Fatal: Compilation aborted
genmath.inc(919,7) Fatal: Internal error 2014051001
Start of compile error summary.
ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.
fpcupdeluxe: Done !!

```

Using FPCUPDeluce 2.4.0h in a UTM VM aarch64 with macOS 15.7.3 on Apple Silicon.

Would appreciate any help and be willing to test whatever you throw at me.

Okay, after the failed attempt to build Lazarus with -WM11.0, here's the output with trying to install the cross compiler for x86_64 Darwin with -WM11.0:

```
/Users/seppel/fpcupdeluxe/fpcsrc/compiler/ppcrossx64 -dFPC_USE_LIBC -Px86_64 -Ur -Xs -O2 -n -Fi../inc -Fi../x86_64 -Fi../unix -Fi../bsd -Fi../bsd/x86_64 -Fi../darwin/x86_64 -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -FE. -FU/Users/seppel/fpcupdeluxe/fpcsrc/rtl/units/x86_64-darwin -Fl/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib -XR/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -vw-n-h-l-d-u-t-p-c- -ap -dx86_64 -dRELEASE -FD/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/usr/bin -XR/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib -Fl/Applications/Xcode.app/Contents/Developer/Platforms/macOSX.platform/Developer/SDKs/macOSX.sdk/usr/lib/system  -Us -Sg ../bsd/system.pp
make[1]: *** [darwin_all] Error 2
make: *** [rtl_all] Error 2
<stdin>:57026:2: error: invalid instruction mnemonic 'movz'
        movz
        ^~~~
<stdin>:57026:6: error: assembler local symbol 'Ld23' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld8' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld16' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld57' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Lj9601' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld45' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld7' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld35' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld4' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld54' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld52' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld53' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld48' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld55' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld49' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld20' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld29' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld15' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld21' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld41' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld10' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld44' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld32' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Lj9610' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld18' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld19' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld11' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld56' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld50' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld12' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld6' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld26' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld14' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld42' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld2' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld30' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld22' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld31' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld40' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld5' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld13' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld36' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld9' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld39' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld33' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld37' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld24' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Lj9612' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld28' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld47' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld25' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld34' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld27' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld3' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld43' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld51' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld17' not defined
        movz
            ^
<stdin>:57026:6: error: assembler local symbol 'Ld46' not defined
        movz
            ^
genmath.inc(606,5) Fatal: Internal error 2014051001
Fatal: Compilation aborted
fpcupdeluxe: ERROR: FPC Cross Installer (BuildModuleCustom: FPC): Running cross compiler fpc /usr/bin/make for x86_64-darwin failed with an error code.


ERROR: Fpcupdeluxe fatal error !
Sequencer (FPCCleanBuildOnly): Failure running fpcupdeluxe: error executing sequence FPCCleanBuildOnly
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
ERROR: Building cross-tools failed. Aborting.
```

--- a/sources/installerfpc.pas
+++ b/sources/installerfpc.pas
@@ -1521,6 +1521,14 @@ begin
                 NativeCompilerOptions:=NativeCompilerOptions+' -d'+DEFINE_SOFT_FPUX80;
               end;
             end;
+          end
+          else if (GetSourceCPU=GetCPU(TCPU.aarch64)) then
+          begin
+            if (CrossInstaller.TargetCPU in [TCPU.i386,TCPU.i8086,TCPU.x86_64]) then
+            begin
+              Infoln(infotext+'Adding -d'+DEFINE_SOFT_FPUX80+' to compiler options to enable 80bit (soft)float support.',etInfo);
+              NativeCompilerOptions:=NativeCompilerOptions+' -d'+DEFINE_SOFT_FPUX80;
+            end;
           end;
           {$endif}
 
@@ -1583,18 +1591,6 @@ begin
             if UseLibc then CrossCompilerOptions:=CrossCompilerOptions+'-d'+DEFINE_FPC_LIBC+' ';
             //if UseLibc then NativeCompilerOptions:=NativeCompilerOptions+' -d'+DEFINE_FPC_LIBC;
             //if UseLibc then CrossInstaller.AddFPCCFGSnippet('-d'+DEFINE_FPC_LIBC);
-
-            if (GetSourceCPU=GetCPU(TCPU.aarch64)) then
-            begin
-              if (CrossInstaller.TargetCPU in [TCPU.i386,TCPU.i8086,TCPU.x86_64]) then
-              begin
-                if (MakeCycle in [st_RtlBuild,st_PackagesBuild]) then
-                begin
-                  Infoln(infotext+'Adding -d'+DEFINE_SOFT_FPUX80+' to compiler cross-options to enable 80bit (soft)float support.',etInfo);
-                  CrossCompilerOptions:=CrossCompilerOptions+'-d'+DEFINE_SOFT_FPUX80+' ';
-                end;
-              end;
-            end;
 
             for i:=0 to CrossInstaller.CrossOpt.Count-1 do
               CrossCompilerOptions:=CrossCompilerOptions+Trim(CrossInstaller.CrossOpt[i])+' ';

